bot/seasons/evergreen/space.py

```diff
@@ -0,0 +1,205 @@
+        # Parse date if provided
+        if date:
+            try:
+                show_date = datetime.strptime(date, "%Y-%m-%d").date().isoformat()
+            except ValueError:
+                await ctx.send(f"Invalid date {date}. Please make sure your date is in format YYYY-MM-DD.")
+                return
+        else:
+            show_date = None
+
+        # Generate URL and make request to API
+        async with self.http_session.get(
+                url=f"{NASA_EPIC_BASE_URL}/api/natural{f'/date/{show_date}' if show_date else ''}"
+        ) as resp:
+            data = await resp.json()
+
+        if len(data) < 1:
+            await ctx.send("Can't find any images in this date.")
+            return
+
+        # Get random item from result that will be shown
+        item = random.choice(data)
+
+        # Split date for image URL
+        year, month, day = item["date"].split(" ")[0].split("-")
+
+        image_url = f"{NASA_EPIC_BASE_URL}/archive/natural/{year}/{month}/{day}/jpg/{item['image']}.jpg"
+
+        # Create embed, fill and send it
+        embed = Embed(title="Earth Image", description=item["caption"])
+        embed.set_image(url=image_url)
+        embed.set_footer(text=f"Identifier: {item['identifier']} \u2022 Powered by NASA API")
+
+        await ctx.send(embed=embed)
+
+    @space.command(name="mars")
+    async def mars(self,
+                   ctx: Context,
+                   date: DateConverter,
+                   rover: Optional[str] = "curiosity"
+                   ) -> None:
+        """
+        Get random Mars image by date. Support both SOL (martian solar day) and earth date and rovers.
+
+        Earth date formatting is YYYY-MM-DD. Current max date is 2019-09-28 and min 2012-08-06.
+        Rovers images dates:
+        - Curiosity -> 2012-08-06 until 2019-09-28
+        - Opportunity -> 2004-01-25 until 2018-06-11
+        - Spirit -> 2004-01-04 until 2010-03-21
+        """
+        # Check does user provided correct rover
+        rover = rover.lower()
+        if rover not in ["curiosity", "opportunity", "spirit"]:
+            await ctx.send(f"Invalid rover `{rover}`. Rovers: `Curiosity`, `Opportunity`, `Spirit`")
+            return
+
+        # Create API request parameters, try to parse date
+        params = {
+            "api_key": Tokens.nasa
+        }
+        if isinstance(date, int):
+            params["sol"] = date
+        else:
+            params["earth_date"] = date.date().isoformat()
+
+        result = await self.fetch_from_nasa(f"mars-photos/api/v1/rovers/{rover}/photos", params)
+
+        # Check for empty result
+        if len(result["photos"]) < 1:
+            err_msg = (
+                f"We can't find result in date "
+                f"{date.date().isoformat() if isinstance(date, datetime) else f'{date} SOL'}.\n"
+                f"**Note:** Dates must match with rover's working dates. Please use `{ctx.prefix}help space mars` to "
```

Comment:
The mission manifest provides a `max_date` value, how about we cache this, say, at most once per day for each rover rather than having to constantly adjust the docstring of the command as more photos are made available (since Curiosity is still an ongoing mission)?